### DISCUSSION TITLE
How to set file mode but ignore content?

### DISCUSSION THREAD

**Discussion by infokiller** [CyberShadow/aconfmgr#156] [Q&A] [answered]
Is it possible?

My first attempt was something like the following

```sh
SetFileProperty /etc/file mode 644
IgnorePath /etc/file
```

But that didn't work (and going through the code it was clear it shouldn't).

BTW, after this failed attempt, it seems `file-owners` was corrupted, causing an error in this line: https://github.com/CyberShadow/aconfmgr/blob/master/src/apply.bash#L658

The error was "src/apply.bash: line 658: file_owners[$file]: bad array subscript", and debugging a bit showed that `$file` and `$package` were empty.
I then kept getting this error in subsequent runs, even after I reverted the config change, since apparently `$tmp_dir` is not cleaned up between runs. Why is this the case? are there caching benefits?

Thanks!

**Comment by CyberShadow** [comment, marked as answer]
> Is it possible?

You could do this by using the content of the file on the system:

```bash
cat /etc/file > "$(CreateFile /etc/file)"
```

> I then kept getting this error in subsequent runs, even after I reverted the config change, since apparently `$tmp_dir` is not cleaned up between runs.

Whoops, `"$tmp_dir"/file-owners` indeed should have been cleared on every run. Fixed in 1c627492f8dedc9c4d42988332ab98657f9b2187.

> Why is this the case? are there caching benefits?

Yes, not clearing `$tmp_dir` enables `--skip-inspection` and `--skip-config`.

**Reply by infokiller** [reply]
Great, thanks a lot!
